lib/src/revset_graph.rs

```diff
@@ -15,31 +15,30 @@
 #![allow(missing_docs)]
 
 use std::collections::{HashMap, HashSet, VecDeque};
-
-use crate::backend::CommitId;
+use std::hash::Hash;
 
 #[derive(Debug, PartialEq, Eq, Clone, Hash)]
-pub struct RevsetGraphEdge {
-    pub target: CommitId,
+pub struct RevsetGraphEdge<N> {
```

Comment:
Since these structs/traits are no longer specific to revsets, should I rename them to eg. `GraphEdge`, and rename the file to `graph`?

Reply:
Sounds reasonable. I'm a little worried that just `GraphEdge` and `graph` is a bit too generic, making it sound like something used by things like the `dag_walk` module. But since I can't really think of a better name, I'd say you should feel free to ignore me :)